tests/Zend/Locale/FormatTest.php

```diff
@@ -1108,10 +1116,13 @@ public function testToNumberWithoutFormatWithPrecision()
     public function testCheckDateFormatDoesNotEmitNoticeWhenNoOptionsAreNotProvided()
     {
         try {
+            $locale = setlocale(LC_ALL, 0); // read current locale
             setlocale(LC_ALL, 'en_US'); // test setup
             Zend_Locale_Format::setOptions(array('date_format' => 'yyyy-MM-dd'));
+            $checkDateFormat = Zend_Locale_Format::checkDateFormat('2011-10-21', array());
+            setlocale(LC_ALL, $locale); // restore previous locale
```

Comment:
should this also be copied to the `catch` block below? 🤔

Reply:
err? what? why?

Reply:
I thought of a case when `Zend_Locale_Format::checkDateFormat` would throw a notice, caught by `} catch ( PHPUnit_Framework_Error_Notice $ex ) {` below... then the locale would not be reverted

Reply:
oh, wow. opened pandora box here. looks like the tests are incomplete and inconsistent.

1) `Zend_LocaleTest` backs up `LC_ALL` in `setUp`:
- https://github.com/zf1s/zf1/blob/8e1040b4a5c147f8751994a597253d9c460b10cd/tests/Zend/LocaleTest.php#L62

and in `tearDown` restores after each test:
- https://github.com/zf1s/zf1/blob/8e1040b4a5c147f8751994a597253d9c460b10cd/tests/Zend/LocaleTest.php#L80-L89


afaik phpunit calls tearDown after each test regardless of the test outcome. so wrapping and restoring locales in LocaleTest regarding `LC_ALL` would not be needed. but any other `LC_*` variable that is altered, need to be restored separately. even LC_ALL overrides all other variables, they still have effect if LC_ALL is unset, so need to restore.

also must note that phpunit 3.7 does not recover from subclasses of `\Error` and `\Throwable`.
ideally, no method should throw those new exceptions if all input is validated :D

2) `Zend_Locale_FormatTest` restores locales if TESTS_ZEND_LOCALE_FORMAT_SETLOCALE is set

- https://github.com/zf1s/zf1/blob/8e1040b4a5c147f8751994a597253d9c460b10cd/tests/Zend/Locale/FormatTest.php#L43-L52

it restores LC_ALL to the value of `TESTS_ZEND_LOCALE_FORMAT_SETLOCALE`. we do not set `TESTS_ZEND_LOCALE_FORMAT_SETLOCALE`.

also, the `Zend_Locale_FormatTest` contains a test that alters other locale settings than `LC_ALL`, so these must be restored.

Reply:
Thank you for detailed info. If the case you mentioned in 2) still makes troubles we may return to it later.